testtools/util.go

```diff
@@ -45,11 +47,17 @@ func NewStoringMockUploader(storage *memory.Storage, deltaDataFolder internal.Da
 
 func CreateMockStorageFolder() storage.Folder {
 	var folder = MakeDefaultInMemoryStorageFolder()
+	meta123 := fmt.Sprintf("{\"finish_time\":\"%s\"}", time.Now().AddDate(0, 0, -2).Format(time.RFC3339))
+	meta456 := fmt.Sprintf("{\"finish_time\":\"%s\"}", time.Now().AddDate(0, 0, -1).Format(time.RFC3339))
+	meta000 := fmt.Sprintf("{\"finish_time\":\"%s\"}", time.Now().Format(time.RFC3339))
 	subFolder := folder.GetSubFolder(utility.BaseBackupPath)
 	subFolder.PutObject("base_123_backup_stop_sentinel.json", &bytes.Buffer{})
+	subFolder.PutObject("base_123/metadata.json", strings.NewReader(meta123))
 	subFolder.PutObject("base_456_backup_stop_sentinel.json", strings.NewReader("{}"))
+	subFolder.PutObject("base_456/metadata.json", strings.NewReader(meta456))
 	subFolder.PutObject("base_000_backup_stop_sentinel.json", &bytes.Buffer{}) // last put
-	subFolder.PutObject("base_123312", &bytes.Buffer{})                        // not a sentinel
+	subFolder.PutObject("base_000/metadata.json", strings.NewReader(meta000))
+	subFolder.PutObject("base_123312", &bytes.Buffer{}) // not a sentinel
```

Comment:
You should test backup-list to check the difference with the old behavior.

Reply:
What exactly do I test? There may be a valid difference. The new listing order may be different from the old one if, for example, for some reason sentinel creation took a long time and it's modification time is significantly different from the backup finish time written in metadata.

Reply:
You should check that the time that you get from getBackupTimeSlices should be equal to values in metafiles and different from the time of last modified in storage. Also time in metafiles has better precision than a time of last-modified. So you need to update the integration test of backup-list to check it.